### PR TITLE
Support for random URL

### PR DESCRIPTION
**Pull request by boastful-squirrel**
Sometimes, I think it's quite convenient to generate random URL, so only people with the right one can crawl the server. Let's be honest, it's very cheap security-wise, but I think there could be some use-cases where people would want this feature.

I'm talking about something like this:

``` 
miniserve -i 192.168.0.1 -r /tmp/myshare
# Serving path /tmp/myshare at http://192.168.0.1:8080/ibe1CYIYqIOuCQ_0lkpWM
```

Currently, the option is `-r  / --random-route`, maybe there is a better choice than this, as `-r` makes me think of "recursive", I might be wrong though.

If this feature isn't wanted, I'll close the PR.

### CODE REVIEW

**Comment by boastful-squirrel** [comment]
On it :)

**Comment by boastful-squirrel** [comment]
I also removed the `-r` short option

File: README.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ Sometimes this is just a more practical and quick way than doing things properly
 
     miniserve --auth joe:123 unreleased-linux-distros/
 
+### Generate a random route so only people with the right URL can crawl the server:
+
+    miniserve -i 192.168.0.1 -r /tmp/myshare
```
Comment (svenstaro) [inline README.md]: Gotta fix this invocation.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ Sometimes this is just a more practical and quick way than doing things properly
...
+    # Serving path /tmp/myshare at http://192.168.0.1:8080/ibe1CYIYqIOuCQ_0lkpWM
```
Comment (svenstaro) [inline README.md]: Also this.

**Comment by mglax** [comment]
Yep, I'll do it  asap. Also, I failed with Git it seems, as all changes done on the fancy-listing branches also appear here. I'll fix this asap

File: src/main.rs [hunk 3]
```diff
@@ -164,6 +177,12 @@ pub fn parse_args() -> MiniserveConfig {
         None
     };
 
+    let mut random_route = None;
```
Comment (svenstaro) [inline src/main.rs]: Can you make this a bit like https://github.com/svenstaro/miniserve/pull/26/files#diff-639fbc4ef05b315af92b4d836c31b023R158?

This way, you don't have the mutable variable around.

File: src/main.rs [hunk 4]
```diff
@@ -136,12 +142,19 @@ pub fn parse_args() -> MiniserveConfig {
                 .long("no-symlinks")
                 .help("Do not follow symbolic links"),
         )
+        .arg(
+            Arg::with_name("random-route")
```
Comment (svenstaro) [inline src/main.rs]: This should conflict with an explicitly chosen route.

File: src/main.rs [hunk 5]
```diff
@@ -198,10 +218,14 @@ fn configure_app(app: App<MiniserveConfig>) -> App<MiniserveConfig> {
         }
     };
 
+    let route_url = "/".to_string();
```
Comment (svenstaro) [inline src/main.rs]: Maybe call this `base_route` to communicate the intention?

Comment (svenstaro) [reply src/main.rs]: And in fact, you can probably just remove this as you don't really need it. Just put it directly in your format string at `let full_route`.

**Comment by boastful-squirrel** [comment]
I'll look into that, thanks for the advice :)

**Comment by boastful-squirrel** [comment]
I'm closing this PR as I messed up hard with git, opening a new one with all requestes changes

**Pull request closed without merging**